pallets/attestation/src/migrations.rs

```diff
@@ -0,0 +1,139 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::{
+	traits::{Get, GetStorageVersion, OnRuntimeUpgrade, ReservableCurrency, StorageVersion},
+	weights::Weight,
+};
+use kilt_support::migration::switch_reserved_to_hold;
+use log;
+use sp_runtime::SaturatedConversion;
+use sp_std::marker::PhantomData;
+
+use crate::{AccountIdOf, Attestations, Config, CurrencyOf, HoldReason, Pallet};
+
+const CURRENT_STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+const TARGET_STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
+
+pub struct BalanceMigration<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for BalanceMigration<T>
+where
+	<T as Config>::Currency: ReservableCurrency<T::AccountId>,
+{
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		log::info!("Attestation: Initiating migration");
+
+		let onchain_storage_version = Pallet::<T>::on_chain_storage_version();
+
+		if onchain_storage_version.eq(&CURRENT_STORAGE_VERSION) {
+			TARGET_STORAGE_VERSION.put::<Pallet<T>>();
```

Comment:
If I understand correctly, the result of this block is 1R and 1W + the result of `do_migration`, but it's not returned, so it is then discared and `<T as frame_system::Config>::DbWeight::get().reads_writes(1, 0)` at the end of the function is returned? I think that's wrong. Either the final weight should `return`ed here, or then you can return the weight without the read, and add the additional read at the end. Same for the other pallets.

Reply:
It is returned :thinking:. Give the block again a look: